packages/openapi-fetch/src/index.d.ts

```diff
@@ -39,6 +39,15 @@ export type QuerySerializer<T> = (
 export type BodySerializer<T> = (body: OperationRequestBodyContent<T>) => any;
 
 export type ParseAs = "json" | "text" | "blob" | "arrayBuffer" | "stream";
+export type ParseAsResponse<T, K extends ParseAs> = K extends "text"
+  ? Awaited<ReturnType<Response["text"]>>
+  : K extends "blob"
```

Comment:
The object was much cleaner, however, I was not able to get that working.

This is messier, but it does have the advantage of _always_ falling back to the original type `T` whereas the cleaner object could return `unknown`